### ISSUE TITLE
CLI version for strings extraction

### ISSUE THREAD

**Issue by Mte90** [vslavik/poedit#399]
Actually the interface has the cool feature of scanning the file and update the po with the new terms.
I want to automatize this part and avoiding to use the interface for it so a command will speed up my job as developer, something that force the file to replace the old string with new will be very useful.

**Comment by vslavik** [comment]
> I want to automatize this part

Use `xgettext`. That's what it's for.

**Comment by Mte90** [comment]
That command support the POedit custom header? If yes sorry for the disturb.

**Comment by vslavik** [comment]
No, it doesn't. If you want to extract from CLI instead of GUI, *don't use these headers* and use the tool that is actually meant for the job. It's not hard — see the GNU gettext manual linked from the Help menu.

**Comment by Mte90** [comment]
My needs is based on the support of the PoEdit headers because in the WordPress world is used everywhere. 
So I will see on the gettext docs when I have time for it but I was hoping on something more fast to include that step on my building process.

**Comment by vslavik** [comment]
> but I was hoping on something more fast to include that step on my building process.

Step 1: Read the `xgettext` man page, because there's probably a reason you're being pointed to it.
Step 2: Realize it is *trivial* to use and, indeed, "fast to include".
Step 3: There's no step 3.

At the moment, I have no plans for Poedit CLI — it would be a non-trivial amount of *pointless* work for something that `xgettext` does already and does better. I'm reopening this as a feature request, but I'll be honest: it's not on the table now.

*(A note for future readers: if you stumbled upon this and it would be useful for you, feel free to upvote; don't add "me too" comments unless there's something substantial to add.)*

**Comment by Mte90** [comment]
I created a python script that do that: https://github.com/CodeAtCode/PoEditZ
Actually it is working only that the po file merged don't keep the old headers, I will do probably something with python to replace that but I don't understand why msgmerge don't keep the others.

**Comment by Mte90** [comment]
I finished that script :-)

**Comment by vslavik** [comment]
@Mte90 if you wish to make a PR *for Poedit*, please do so, it would be very welcome. However please don't fill this place, which is a *Poedit* bug tracker, with off-topic content such as commentary about your progress on your own pet project.

P.S. The name is "Poedit", not "PoEdit", "POEdit" or whatever else.

**Comment by mwpowellhtx** [comment]
Note the goal is to include those targets in a contemporary `csproj` project. FWIW, the following seems to be working pretty well.

```xml
<ItemGroup>
    <GettextSource Include="Locale\da-DK\LC_MESSAGES\*.po" />
    <GettextSource Include="Locale\de-DE\LC_MESSAGES\*.po" />
</ItemGroup>

<Target Name="GettextMsgFmt" BeforeTargets="Build">
    <PropertyGroup>
        <MsgFmtDir>C:\Program Files (x86)\Poedit\GettextTools\bin\</MsgFmtDir>
        <MsgFmtExe>msgfmt.exe</MsgFmtExe>
        <MsgFmtPath>$(MsgFmtDir)$(MsgFmtExe)</MsgFmtPath>
        <MsgFmtInputExtension>.po</MsgFmtInputExtension>
        <MsgFmtOutputExtension>.mo</MsgFmtOutputExtension>
    </PropertyGroup>
    <ItemGroup>
        <GettextSource>
            <_Src>%(FileName)$(MsgFmtInputExtension)</_Src>
            <_FullSrc>%(Directory)%(FileName)$(MsgFmtInputExtension)</_FullSrc>
            <_Dest>%(FileName)$(MsgFmtOutputExtension)</_Dest>
            <_FullDest>%(Directory)\%(FileName)$(MsgFmtOutputExtension)</_FullDest>
        </GettextSource>
    </ItemGroup>
    <Message Importance="high" Text="Building Gettext file '%(GettextSource._FullSrc)' to '%(GettextSource._FullDest)' ..." />
    <Exec Command='"$(MsgFmtPath)" -o %(GettextSource._Dest) %(GettextSource._Src)' WorkingDirectory="%(GettextSource.RootDir)%(GettextSource.Directory)" ContinueOnError="false" />
</Target>
```

After you get past the `<Target/>` scaffold, the key is here: `"$(MsgFmtPath)" -o %(GettextSource._Dest) %(GettextSource._Src)`.

**Comment by mwpowellhtx** [comment]
Apologies for the comment confusion, this one was a good starting point for me after all. Thank you...
